tactic/make_lemma.lean

```diff
@@ -0,0 +1,43 @@
+/-
+Copyright (c) 2017 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+
+import data.buffer.parser
+
+open lean.parser tactic interactive parser
+
+@[user_attribute] meta def field_lemma_attr : user_attribute :=
+{ name := `field_lemma, descr := "This definition was automatically generated from a structure field by `make_lemma`." }
+
+
+/- Make lemma takes a structure field, and makes a new, definitionally simplified copy of it, appending `_lemma` to the name.
+   The main application is to provide clean versions of structure fields that have been tagged with an auto_param. -/
+meta def make_lemma (d : declaration) (new_name : name) : tactic unit :=
```

Comment:
The name of this tactic is extremely generic. How about `mk_structure_lemma` or `restate_axiom`?

Reply:
Done: `restate_axiom`.